aten/src/ATen/native/LossNLL.cpp

```diff
@@ -402,6 +411,64 @@ Tensor nll_loss(const Tensor & self, const Tensor & target, const Tensor & weigh
   return std::get<0>(at::nll_loss_forward(self, target, weight, reduction, ignore_index));
 }
 
+Tensor nll_loss_nd(const Tensor & self, const Tensor & target, const Tensor & weight, int64_t reduction, int64_t ignore_index) {
```

Comment:
I am open to a better name than this :) Currently we have nll_loss and nll_loss_nd and I do not want to break compatibility.